lib/matplotlib/axes/_axes.py

```diff
@@ -4104,112 +4095,132 @@ def hexbin(self, x, y, C=None, gridsize=100, bins=None,
         specified, it must also be a 1-D sequence of the same length
         as *x* and *y*.)
 
-        *x*, *y* and/or *C* may be masked arrays, in which case only
-        unmasked points will be plotted.
-
-        Optional keyword arguments:
-
-        *gridsize*: [ 100 | integer ]
-           The number of hexagons in the *x*-direction, default is
-           100. The corresponding number of hexagons in the
-           *y*-direction is chosen such that the hexagons are
-           approximately regular. Alternatively, gridsize can be a
-           tuple with two elements specifying the number of hexagons
-           in the *x*-direction and the *y*-direction.
-
-        *bins*: [ *None* | 'log' | integer | sequence ]
-           If *None*, no binning is applied; the color of each hexagon
-           directly corresponds to its count value.
-
-           If 'log', use a logarithmic scale for the color
-           map. Internally, :math:`log_{10}(i+1)` is used to
-           determine the hexagon color.
-
-           If an integer, divide the counts in the specified number
-           of bins, and color the hexagons accordingly.
-
-           If a sequence of values, the values of the lower bound of
-           the bins to be used.
-
-        *xscale*: [ 'linear' | 'log' ]
-           Use a linear or log10 scale on the horizontal axis.
-
-        *yscale*: [ 'linear' | 'log' ]
-           Use a linear or log10 scale on the vertical axis.
+        Parameters
+        ----------
+        x : array
+            may be masked array, in which case only unmasked points
+            will be plotted
```

Comment:
@QuLogic has the right suggestion below, applicable here also: group `x, y : array or masked array`.  `C` needs its own line because it is optional.